From 7407c483b9ab89d675784e1636308d526144004a Mon Sep 17 00:00:00 2001
From: Sam Thursfield <sam.thursfield@codethink.co.uk>
Date: Wed, 20 May 2026 11:45:34 +0100
Subject: [PATCH 01/20] FDSDK 26.08: Update junction and remove obsolete
 patches

Remove patches that are applied upstream in FDSDK master
---
 elements/freedesktop-sdk.bst                  |  4 +-
 ...03-openssh-Use-etc-ssh-as-sysconfdir.patch | 56 ------------------
 ...-openssh-Include-ssh-_config.d-.conf.patch | 53 -----------------
 ...-Disable-event-activation-by-default.patch | 53 -----------------
 .../0007-Update-Pipewire-to-1.6.7.patch       | 57 ++++---------------
 5 files changed, 14 insertions(+), 209 deletions(-)
 delete mode 100644 patches/freedesktop-sdk/0003-openssh-Use-etc-ssh-as-sysconfdir.patch
 delete mode 100644 patches/freedesktop-sdk/0004-openssh-Include-ssh-_config.d-.conf.patch
 delete mode 100644 patches/freedesktop-sdk/0005-lvm2-Disable-event-activation-by-default.patch

diff --git a/elements/freedesktop-sdk.bst b/elements/freedesktop-sdk.bst
index d025c59a0..05acf206c 100644
--- a/elements/freedesktop-sdk.bst
+++ b/elements/freedesktop-sdk.bst
@@ -3,8 +3,8 @@ kind: junction
 sources:
 - kind: git_repo
   url: gitlab:freedesktop-sdk/freedesktop-sdk.git
-  track: freedesktop-sdk-25.08*
-  ref: freedesktop-sdk-25.08.13-0-g8446990f0a549bb1f3ceb654af64fc176b274488
+  track: master
+  ref: freedesktop-sdk-25.08rc.3-2101-g69c67eb9198d85a7eed56664c0d9ede6dcf5081e
 - kind: patch_queue
   path: patches/freedesktop-sdk
 
diff --git a/patches/freedesktop-sdk/0003-openssh-Use-etc-ssh-as-sysconfdir.patch b/patches/freedesktop-sdk/0003-openssh-Use-etc-ssh-as-sysconfdir.patch
deleted file mode 100644
index fcbfdb5af..000000000
--- a/patches/freedesktop-sdk/0003-openssh-Use-etc-ssh-as-sysconfdir.patch
+++ /dev/null
@@ -1,56 +0,0 @@
-From cb7355848d5632c17355078ae33508f7766409f8 Mon Sep 17 00:00:00 2001
-From: Valentin David <me@valentindavid.com>
-Date: Sun, 26 Oct 2025 11:03:32 +0100
-Subject: [PATCH 3/7] openssh: Use /etc/ssh as sysconfdir
-
-All other distros use that. And there are expectations from systemd
-for it to be configured so.
-
-Also fix a /var/run
----
- elements/components/openssh.bst | 2 ++
- 1 file changed, 2 insertions(+)
-
-diff --git a/elements/components/openssh.bst b/elements/components/openssh.bst
-index a7c0361f3..2ccc03748 100644
---- a/elements/components/openssh.bst
-+++ b/elements/components/openssh.bst
-@@ -9,33 +9,35 @@ build-depends:
- depends:
- - bootstrap/libxcrypt.bst
- - public-stacks/runtime-minimal.bst
- - components/systemd-libs.bst # libfido2 depends on udev
- 
- runtime-depends:
- - components/grep.bst
- - components/sed.bst
- 
- variables:
-+  sysconfdir: '/etc/ssh'
-   arch_options: ''
-   (?):
-   # ppc64le doesn't support --fzero-call-used-regs=used flag
-   # which is part of openssh 'hardening' and the rest have marginal importance
-   # https://gitweb.gentoo.org/repo/gentoo.git/commit/net-misc/openssh?id=0b22d07f89b16ac3400e45077702ac4c4492e5a4
-   - target_arch == "ppc64le":
-       arch_options: "--without-hardening"
-   # linux-pam provides /etc/pam.d/sshd, so this is the pam service name
-   conf-local: >-
-     --with-mantype=man
-     --with-pam
-     --with-pam-service=sshd
-     --without-zlib-version-check
-+    --with-pid-dir=/run
-     --with-security-key-builtin
-     %{arch_options}
- 
- config:
-   install-commands:
-     (>):
-     - |
-       sed 's/#UsePAM.*/UsePAM yes/' -i "%{install-root}%{sysconfdir}/sshd_config"
-       install -Dm755 contrib/ssh-copy-id -t "%{install-root}%{bindir}"
-       install -Dm644 contrib/ssh-copy-id.1 -t "%{install-root}%{mandir}/man1"
--- 
-2.54.0
-
diff --git a/patches/freedesktop-sdk/0004-openssh-Include-ssh-_config.d-.conf.patch b/patches/freedesktop-sdk/0004-openssh-Include-ssh-_config.d-.conf.patch
deleted file mode 100644
index 6e8e27abe..000000000
--- a/patches/freedesktop-sdk/0004-openssh-Include-ssh-_config.d-.conf.patch
+++ /dev/null
@@ -1,53 +0,0 @@
-From deba018b7180e08cb01685e422b7b4fc668d5ef4 Mon Sep 17 00:00:00 2001
-From: Valentin David <me@valentindavid.com>
-Date: Sun, 26 Oct 2025 16:06:02 +0100
-Subject: [PATCH 4/7] openssh: Include ssh*_config.d/*.conf
-
-Most distributions include system-wide configuration fragments from
-and some software expect it: systemd, libvirt.
----
- elements/components/openssh.bst | 14 ++++++++++++++
- 1 file changed, 14 insertions(+)
-
-diff --git a/elements/components/openssh.bst b/elements/components/openssh.bst
-index 2ccc03748..d48247cd1 100644
---- a/elements/components/openssh.bst
-+++ b/elements/components/openssh.bst
-@@ -28,20 +28,34 @@ variables:
-   conf-local: >-
-     --with-mantype=man
-     --with-pam
-     --with-pam-service=sshd
-     --without-zlib-version-check
-     --with-pid-dir=/run
-     --with-security-key-builtin
-     %{arch_options}
- 
- config:
-+  configure-commands:
-+    (<):
-+    - |
-+      cat <<EOF >>"sshd_config"
-+
-+      Include %{sysconfdir}/sshd_config.d/*.conf
-+      EOF
-+
-+    - |
-+      cat <<EOF >>"ssh_config"
-+
-+      Include %{sysconfdir}/ssh_config.d/*.conf
-+      EOF
-+
-   install-commands:
-     (>):
-     - |
-       sed 's/#UsePAM.*/UsePAM yes/' -i "%{install-root}%{sysconfdir}/sshd_config"
-       install -Dm755 contrib/ssh-copy-id -t "%{install-root}%{bindir}"
-       install -Dm644 contrib/ssh-copy-id.1 -t "%{install-root}%{mandir}/man1"
- 
- public:
-   bst:
-     split-rules:
--- 
-2.54.0
-
diff --git a/patches/freedesktop-sdk/0005-lvm2-Disable-event-activation-by-default.patch b/patches/freedesktop-sdk/0005-lvm2-Disable-event-activation-by-default.patch
deleted file mode 100644
index 3013de9f2..000000000
--- a/patches/freedesktop-sdk/0005-lvm2-Disable-event-activation-by-default.patch
+++ /dev/null
@@ -1,53 +0,0 @@
-From e51fde61ed09195ba684340c0b3be5f4c6fd9cad Mon Sep 17 00:00:00 2001
-From: Valentin David <me@valentindavid.com>
-Date: Sat, 24 Jan 2026 12:13:27 +0100
-Subject: [PATCH 5/7] lvm2: Disable event activation by default
-
-Because lvm2 does not have a distro wide configuration in /usr and all
-is in /etc, configuring it is not compatible with systemd-sysupdate
-OSes - because systemd-udevd starts before systemd-tmpfiles. Also on
-GNOME OS, installation is confusing for the user that cannot format a
-disk before they deactivate the volume groups taking phyisical volumes
-from that disk.
-
-The solution is to disable this by default
-
-Now, Bluefin users will be fine with lvm2 installations, as
-bootc/OSTree still provides /etc at the right time. So for those
-distros we will set the initial value in /etc to activate the groups.
-
-It does not seem Endless OS uses lvm2 volume groups. Furthermore, it
-could also set the configuration in /etc since it uses OSTree.
----
- elements/components/lvm2-base.bst | 1 +
- 1 file changed, 1 insertion(+)
-
-diff --git a/elements/components/lvm2-base.bst b/elements/components/lvm2-base.bst
-index 5f289a56a..5b24218eb 100644
---- a/elements/components/lvm2-base.bst
-+++ b/elements/components/lvm2-base.bst
-@@ -15,20 +15,21 @@ variables:
-   conf-local: >-
-     --enable-dmeventd
-     --enable-cmdlib
-     --enable-pkgconfig
-     --enable-udev_sync
-     --enable-udev_rules
-     --with-confdir=%{sysconfdir}
-     --with-staticdir=%{sbindir}
-     --with-usrlibdir=%{libdir}
-     --with-usrsbindir=%{sbindir}
-+    --with-default-event-activation=0
- 
- config:
-   configure-commands:
-     (<):
-     - |
-       rm configure
- 
- public:
-   bst:
-     split-rules:
--- 
-2.54.0
-
diff --git a/patches/freedesktop-sdk/0007-Update-Pipewire-to-1.6.7.patch b/patches/freedesktop-sdk/0007-Update-Pipewire-to-1.6.7.patch
index ebb422e55..5a4eb7998 100644
--- a/patches/freedesktop-sdk/0007-Update-Pipewire-to-1.6.7.patch
+++ b/patches/freedesktop-sdk/0007-Update-Pipewire-to-1.6.7.patch
@@ -1,54 +1,21 @@
-From 34bfea9f255123a4c828b1697fe0af53f765365e Mon Sep 17 00:00:00 2001
-From: freedesktop_sdk_updater <freedesktop_sdk_updater@libreml.com>
-Date: Thu, 19 Feb 2026 17:09:11 +0000
+From 7692238e17543ba582c17d911e3d1cb8dec7b6b3 Mon Sep 17 00:00:00 2001
+From: Hec <hec@heccraft.com>
+Date: Fri, 19 Jun 2026 19:00:26 -0400
 Subject: [PATCH 7/7] Update Pipewire to 1.6.7
 
 ---
- elements/components/pipewire-base.bst | 7 ++++++-
- 1 file changed, 6 insertions(+), 1 deletion(-)
+ elements/components/_private/pipewire-base.bst | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
 
-diff --git a/elements/components/pipewire-base.bst b/elements/components/pipewire-base.bst
-index a09759fdf..9b4ee6c76 100644
---- a/elements/components/pipewire-base.bst
-+++ b/elements/components/pipewire-base.bst
-@@ -49,20 +49,24 @@ variables:
-     -Dlibcanberra=disabled
-     -Dbluez5-codec-lc3plus=disabled
-     -Dlv2=disabled
-     -Dlibjack-path=%{libdir}
-     -Djack-devel=true
-     -Dudevrulesdir=$(pkg-config --variable=udevdir udev)/rules.d
-     -Dsession-managers=[]
-     -Dlibffado=disabled
-     -Dsdl2=disabled
-     -Dsnap=disabled
-+    -Dfftw=disabled
-+    -Dbluez5-codec-ldac-dec=disabled
-+    -Dbluez5-plc-spandsp=disabled
-+    -Donnxruntime=disabled
- 
- config:
-   install-commands:
-     (>):
-     - |
-       for module in '%{install-root}%{libdir}'/pipewire-0.3/*.so; do
-         if ! objdump -x "${module}" | grep -q 'NEEDED *libpipewire-module-'; then
-           patchelf --remove-rpath "${module}"
-         fi
-       done
-@@ -135,11 +139,12 @@ public:
-       - '%{debugdir}%{libdir}/pipewire-0.3/libpipewire-module-spa-node-factory.so.debug'
-       - '%{debugdir}%{libdir}/pipewire-0.3/libpipewire-module-spa-node.so.debug'
-       - '%{debugdir}%{libdir}/pipewire-0.3/libpipewire-module-zeroconf-discover.so.debug'
- 
-   cpe:
-     product: pipewire
- 
- sources:
+diff --git a/elements/components/_private/pipewire-base.bst b/elements/components/_private/pipewire-base.bst
+index 781cebb62..35eb70ab2 100644
+--- a/elements/components/_private/pipewire-base.bst
++++ b/elements/components/_private/pipewire-base.bst
+@@ -147,4 +147,4 @@ sources:
  - kind: git_repo
    url: freedesktop:PipeWire/pipewire.git
--  ref: 1.4.9-0-gfd60e04525f3a04d90bf50085222e0cc9139b4a4
-+  track: "*.*.*"
+   track: '*.[02468].*'
+-  ref: 1.6.6-0-ga1c1d9f9f128ec987f23a0140a1704b8f8781787
 +  ref: 1.6.7-0-g3b2cb4fb037bf6033b87d3c87ee917b2f686d309
 -- 
 2.54.0
-- 
2.54.0

